component/src/main/java/org/wso2/extension/siddhi/map/avro/sourcemapper/AvroSourceMapper.java

```diff
@@ -0,0 +1,424 @@
+                                        "avro message should be specified as quoted json string.",
+                        type = {DataType.STRING})
+        },
+        examples = {
+                @Example(
+                        syntax = "@source(type='inMemory', topic='user', @map(type='avro', schema .def = \"\"\"" +
+                                "{\"type\":\"record\",\"name\":“userInfo\",\"namespace\":\"user.example\",\"fields\":" +
+                                "[{\"name\":\"name\",\"type\":“string”}, {\"name\":\"age\", \"type\":\"int\"}]}" +
+                                "\"\"\"))\n"
+                                + "define stream userStream (name string, age int );\n",
+                        description = "Above configuration will do a default Avro input mapping. " +
+                                "The input avro message containing user info will be " +
+                                "converted  to a siddhi event.\n "
+                                + "Expected input is a byte array.")
+        }
+)
+
+public class AvroSourceMapper extends SourceMapper {
+
+    private static final Logger log = Logger.getLogger(AvroSourceMapper.class);
+    private static final String DEFAULT_AVRO_MAPPING_PREFIX = "schema";
+    private static final String SCHEMA_IDENTIFIER = "def";
+    private static final String DEFAULT_JSON_PATH = "$";
+
+    private StreamDefinition streamDefinition;
+    private List<Attribute> streamAttributes;
+    private int streamAttributesSize;
+    private JsonFactory jsonFactory;
+    private AttributeConverter attributeConverter = new AttributeConverter();
+    private ObjectMapper objectMapper = new ObjectMapper();
+    private Gson gson = new Gson();
+    private Schema schema;
+
+    /**
+     * Initialize the mapper and the mapping configurations.
+     *
+     * @param streamDefinition
+     * @param optionHolder
+     * @param attributeMappingList
+     * @param configReader
+     * @param siddhiAppContext
+     */
+    @Override
+    public void init(StreamDefinition streamDefinition, OptionHolder optionHolder, List<AttributeMapping>
+            attributeMappingList, ConfigReader configReader, SiddhiAppContext siddhiAppContext) {
+        jsonFactory = new JsonFactory();
+        this.streamDefinition = streamDefinition;
+        this.streamAttributes = this.streamDefinition.getAttributeList();
+        this.streamAttributesSize = this.streamDefinition.getAttributeList().size();
+        schema = getAvroSchema(optionHolder.validateAndGetStaticValue(DEFAULT_AVRO_MAPPING_PREFIX.concat(".").
+                        concat(SCHEMA_IDENTIFIER),
+                null));
+    }
+
+    private Schema getAvroSchema(String schemaDefinition) {
+        if (schemaDefinition != null) {
+            return new Schema.Parser().parse(schemaDefinition);
+        } else {
+            log.error("Avro Schema is not specified in the stream definition");
```

Comment:
Do we allow to continue after this error? Don't we need to throw the error?

Reply:
Yes, the error need to be thrown, but at this moment it is done, since this method will be changed in future implementation. This PR refers only to one use case. Hopefully will be changing the method to throw the error in future PRs created by next week.